### ISSUE TITLE
Rename stage 5-app-infra for internal consistency on pipeline layers (foundation vs infra vs application)

### ISSUE THREAD

**Issue by eeaton** [terraform-google-modules/terraform-example-foundation#1287]
### TL;DR

Terminology and concepts are not used consistently for different pipeline layers, and it breaks down at stage 5-app-infra. This should be renamed to help clarify how the enterprise foundation blueprint has modular support for other blueprints, particularly the enterprise application blueprint.

The [enterprise foundations blueprint](https://cloud.google.com/architecture/security-foundations/deployment-methodology#pipeline-layers) guide that describes the reference architecture and recommendations deployed by this repo states the following about [pipeline layers](https://cloud.google.com/static/architecture/security-foundations/images/example-access-patterns.svg):

- Stages 0-3 are considered part of the **foundation pipeline**
- Stages 4-5 are considered part of the **infrastructure pipeline**
- The scope of the enterprise foundation blueprint does not address an **application pipeline** in detail; it introduces the concept but explicitly descopes it, because many workload teams might have different pipelines for their software development lifecycle. It instead links to a separate asset, the [enterprise applicaiton blueprint](https://cloud.google.com/architecture/enterprise-application-blueprint/architecture) which covers recommendations for such pipelines in great detail.

This repo creates confusion by calling stage 5 as `5-app-infra`, suggesting it is an example of an application pipeline.  We can make the repo easier to use and understand by being explicit about what it covers and what it does not, we should avoid suggesting this example foundation covers application pipeline or internal developer platform capabilities.

### Detailed design

```markdown
A spot check of some references in code to update (not necessarily exhaustive)
 - 5-app-infra (directory) --> 5-app-infra
 - app_infra_test.go --> project_infra_test.go
 - TestAppInfra  --> testProjectInfra
 - appInfra --> projectInfra
 - terraform/app-infra/business_unit_1/development --> terraform/project-infra/business_unit_1/development
 - "UPDATE_APP_INFRA_BUCKET"
 - gcp-policies-app-infra
 - AppInfraRepo
```
